src/Video.js

```diff
@@ -742,6 +743,9 @@ export default class Video extends React.Component {
 
   togglePaused = (pausedOverwrite, skipActionOnCasting) => {
     this.updateVideoProgress();
+    if (this.state.showPoster){
```

Comment:
You should combine the 2 setstates here, because all setstates re-render the component, so something like this.setState(state => ({paused, showPoster: state.showPoster ? false : true }))

Reply:
Done!